vortex-tui/src/browse/mod.rs

```diff
@@ -169,17 +169,29 @@ fn handle_search_mode(app: &mut AppState, event: Event) -> HandleResult {
                 //
                 // We can eliminate the search filter when we do this
                 if app.current_tab == Tab::Layout && app.cursor.layout().nchildren() > 0 {
-                    // Descend into the layout subtree for the selected child.
-                    let selected = app.layouts_list_state.selected().unwrap_or_default();
-                    app.cursor = app.cursor.child(selected);
+                    // Descend into the layout subtree for the selected child, do nothing if there's nothing to select.
+                    if let Some(selected) = app.layouts_list_state.selected() {
+                        app.cursor = match app.filter.as_ref() {
+                            None => app.cursor.child(selected),
+                            Some(filter) => {
+                                let child_idx = filter
+                                    .iter()
+                                    .enumerate()
+                                    .filter_map(|(idx, show)| show.then_some(idx))
+                                    .nth(selected)
+                                    .vortex_expect("There must be a selected item in the filter");
```

Comment:
I don't think we want to panic if nothing matches the filter

Reply:
we won't because there must be a selected element at this point